mParticleCore.brs

```diff
@@ -589,7 +589,17 @@ function mParticleStart(options as object, messagePort as object)
         end function,
 
         isEmpty: function(input as dynamic) as boolean
-            return input = invalid or len(input) = 0
+            if input = invalid
+                return true
+            else
+                if (LCase(type(input)) = "roarray" or LCase(type(input)) = "array")
+                    return input[0] = invalid
```

Comment:
I don't think the first element being invalid is indicative of empty. as an example - i'd consider this non-empty:

```
favoriteFood = [invalid, "indian", "italian"]
```

how about using the Count() property of ifarray? https://developer.roku.com/docs/references/brightscript/interfaces/ifarray.md

```

Reply:
Count() was giving me other issues but I'll look at it again